### PR TITLE
event handlers: improve docs

### PR DESCRIPTION
**Pull request by oliver-sanders**
**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.

Goes with https://github.com/cylc/cylc-flow/pull/5655

### CODE REVIEW

File: src/user-guide/writing-workflows/runtime.rst [hunk 1]
```diff
@@ -634,18 +637,40 @@ can be:
...
-are triggered.
+.. cylc-scope:: flow.cylc[runtime][<namespace>][events]
+
+Alternatively you can configure a list of generic event :cylc:conf:`handlers` to be run
+for configured :cylc:conf:`handler events`.
+
+:cylc:conf:`handler events`
+   A list of events which may include any of the above
+   events (e.g. ``submission failed`` or ``warning``) or
+   any of a task's :term:`custom outputs <custom output>`.
```
Comment (oliver-sanders) [inline src/user-guide/writing-workflows/runtime.rst:654]: This is piece of information which was missing before.

Turns out custom outputs can be used in `handler events`, this is covered by an existing functional test.

File: src/glossary.rst [hunk 2]
```diff
@@ -1654,8 +1654,6 @@ Glossary
       WARNING messages from a task :term:`job`.
 
 
-   .. TODO cylc-flow cfgspec/workflow.py references "event handlers" plural
```
Comment (oliver-sanders) [inline src/glossary.rst:1657]: The `event handlers` config is plural so this seems legit to me.

**Review by wxtim — APPROVED** [review]
Yes please - I find these settings slightly hard to think about and this documentation update helps.


Probably out of scope...
 I think that there is ambiguity because "handlers" is plural because it might be running multiple scripts, whilst "event handlers" is plural because it might run on multiple events. I wonder if `common handlers` and `common events` might be better terms. Or at least `events handlers`?

**Comment by oliver-sanders** [comment]
I think we might consider a rethink of how we configure event handlers in light of https://github.com/cylc/cylc-flow/issues/5652.

It would make more sense to configure events and handlers as groups e.g. for "failed and submit-failed" events, run handlers x, y, & z, but for "succeeded and info" events, run handler a.

**Pull request merged**